allennlp/data/dataset_readers/reading_comprehension/util.py

```diff
@@ -369,3 +369,41 @@ def handle_cannot(reference_answers: List[str]):
     else:
         reference_answers = [x for x in reference_answers if x != 'CANNOTANSWER']
     return reference_answers
+
+
+def split_token_by_hyphen(token, hyphen):
```

Comment:
Looks like this function is supposed to work for any delimiter. `split_token_by_delimiter` seems more appropriate.

Reply:
Also add type hints for arguments and return values.